std/sumtype.d

```diff
@@ -2642,6 +2642,403 @@ version (D_Exceptions)
     }));
 }
 
+/**
+ * Checks whether a `SumType` contains a value of a given type.
+ *
+ * The types must match exactly, without implicit conversions.
+ *
+ * Params:
+ *   T = the type to check for.
+ */
+template has(T)
+{
+    /**
+     * The actual `has` function.
+     *
+     * Params:
+     *   self = the `SumType` to check.
+     *
+     * Returns: true if `self` contains a `T`, otherwise false.
+     */
+    bool has(Self)(auto ref Self self)
+    if (isSumType!Self)
+    {
+        return self.match!checkType;
+    }
+
+    // Helper to avoid redundant template instantiations
+    bool checkType(Value)(ref Value value)
```

Comment:
Shouldn't this be private given that it's not something that anyone should actual be calling directly?

Reply:
As far as I know, if a template has an eponymous member, it's impossible to access any of its other members, whether they're private or not. But making it private definitely can't hurt, so yes, it should be.